integreat_cms/cms/views/content_revision_view.py

```diff
@@ -0,0 +1,363 @@
+import logging
+
+from django.contrib import messages
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+from django.shortcuts import redirect, render
+from django.urls import reverse
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+from ..constants import status
+from ..decorators import permission_required
+from ..models import Language, Page, POI
+from .pages.page_context_mixin import PageContextMixin
+from .pois.poi_context_mixin import POIContextMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ContentRevisionView(TemplateView):
+    """
+    View for browsing the page/poi revisions and restoring old page/poi revisions
+    """
+
+    #: The template to render (see :class:`~django.views.generic.base.TemplateResponseMixin`)
+    template_name = "content_revisions.html"
+    #: The context dict passed to the template (see :class:`~django.views.generic.base.ContextMixin`)
+
+    def get_context(self, request, *args, **kwargs):
+        """
+        Returns the context of current content
+
+        To be implemented in the inheriting class
+        """
+        raise NotImplementedError
+
+    def can_change_content(self, request, content):
```

Comment:
I like this idea, however there's already an inbuilt Mixin that handles permission in class-based views: [`PermissionRequiredMixin`](https://docs.djangoproject.com/en/4.2/topics/auth/default/#the-permissionrequiredmixin-mixin)